pkg/spanfilter/config/config.go

```diff
@@ -0,0 +1,83 @@
+package config
+
+import (
+	"fmt"
+
+	"github.com/grafana/tempo/pkg/traceql"
+)
+
+type FilterPolicy struct {
+	Include *PolicyMatch `yaml:"include"`
+	Exclude *PolicyMatch `yaml:"exclude"`
+}
+
+type MatchType string
+
+const (
+	Strict MatchType = "strict"
+	Regex  MatchType = "regex"
+)
+
+var (
+	supportedIntrinsics = []traceql.Intrinsic{
+		traceql.IntrinsicKind,
+		traceql.IntrinsicName,
+		traceql.IntrinsicStatus,
+	}
+)
+
+type PolicyMatch struct {
+	MatchType  MatchType              `yaml:"match_type"`
+	Attributes []MatchPolicyAttribute `yaml:"attributes"`
+}
+
+type MatchPolicyAttribute struct {
+	Key   string      `yaml:"key"`
+	Value interface{} `yaml:"value"`
+}
+
+func ValidateFilterPolicy(policy FilterPolicy) error {
+	if policy.Include == nil && policy.Exclude == nil {
+		return fmt.Errorf("invalid filter policy: %v", policy)
+	}
+
+	if policy.Include != nil {
+		if err := ValidatePolicyMatch(policy.Include); err != nil {
+			return err
+		}
+	}
+
+	if policy.Exclude != nil {
+		if err := ValidatePolicyMatch(policy.Exclude); err != nil {
+			return err
```

Comment:
similar nit here. may want to wrap with the policy that failed to match for clarity.